contrib/s2geometry-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+    "${S2_SOURCE_DIR}/s2contains_vertex_query.cc"
+    "${S2_SOURCE_DIR}/s2convex_hull_query.cc"
+    "${S2_SOURCE_DIR}/s2coords.cc"
+    "${S2_SOURCE_DIR}/s2crossing_edge_query.cc"
+    "${S2_SOURCE_DIR}/s2debug.cc"
+    "${S2_SOURCE_DIR}/s2earth.cc"
+    "${S2_SOURCE_DIR}/s2edge_clipping.cc"
+    "${S2_SOURCE_DIR}/s2edge_crosser.cc"
+    "${S2_SOURCE_DIR}/s2edge_crossings.cc"
+    "${S2_SOURCE_DIR}/s2edge_distances.cc"
+    "${S2_SOURCE_DIR}/s2edge_tessellator.cc"
+    "${S2_SOURCE_DIR}/s2error.cc"
+    "${S2_SOURCE_DIR}/s2furthest_edge_query.cc"
+    "${S2_SOURCE_DIR}/s2latlng.cc"
+    "${S2_SOURCE_DIR}/s2latlng_rect.cc"
+    "${S2_SOURCE_DIR}/s2latlng_rect_bounder.cc"
+    "${S2_SOURCE_DIR}/s2lax_loop_shape.cc"
+    "${S2_SOURCE_DIR}/s2lax_polygon_shape.cc"
+    "${S2_SOURCE_DIR}/s2lax_polyline_shape.cc"
+    "${S2_SOURCE_DIR}/s2loop.cc"
+    "${S2_SOURCE_DIR}/s2loop_measures.cc"
+    "${S2_SOURCE_DIR}/s2measures.cc"
+    "${S2_SOURCE_DIR}/s2metrics.cc"
+    "${S2_SOURCE_DIR}/s2max_distance_targets.cc"
+    "${S2_SOURCE_DIR}/s2min_distance_targets.cc"
+    "${S2_SOURCE_DIR}/s2padded_cell.cc"
+    "${S2_SOURCE_DIR}/s2point_compression.cc"
+    "${S2_SOURCE_DIR}/s2point_region.cc"
+    "${S2_SOURCE_DIR}/s2pointutil.cc"
+    "${S2_SOURCE_DIR}/s2polygon.cc"
+    "${S2_SOURCE_DIR}/s2polyline.cc"
+    "${S2_SOURCE_DIR}/s2polyline_alignment.cc"
+    "${S2_SOURCE_DIR}/s2polyline_measures.cc"
+    "${S2_SOURCE_DIR}/s2polyline_simplifier.cc"
+    "${S2_SOURCE_DIR}/s2predicates.cc"
+    "${S2_SOURCE_DIR}/s2projections.cc"
+    "${S2_SOURCE_DIR}/s2r2rect.cc"
+    "${S2_SOURCE_DIR}/s2region.cc"
+    "${S2_SOURCE_DIR}/s2region_term_indexer.cc"
+    "${S2_SOURCE_DIR}/s2region_coverer.cc"
+    "${S2_SOURCE_DIR}/s2region_intersection.cc"
+    "${S2_SOURCE_DIR}/s2region_union.cc"
+    "${S2_SOURCE_DIR}/s2shape_index.cc"
+    "${S2_SOURCE_DIR}/s2shape_index_buffered_region.cc"
+    "${S2_SOURCE_DIR}/s2shape_index_measures.cc"
+    "${S2_SOURCE_DIR}/s2shape_measures.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_build_polygon_boundaries.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_coding.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_contains_brute_force.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_edge_iterator.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_get_reference_point.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_range_iterator.cc"
+    "${S2_SOURCE_DIR}/s2shapeutil_visit_crossing_edge_pairs.cc"
+    "${S2_SOURCE_DIR}/s2text_format.cc"
+    "${S2_SOURCE_DIR}/s2wedge_relations.cc"
+    "${S2_SOURCE_DIR}/strings/ostringstream.cc"
+    "${S2_SOURCE_DIR}/strings/serialize.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/base/dynamic_annotations.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/base/internal/raw_logging.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/base/internal/throw_delegate.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/numeric/int128.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/strings/ascii.cc"
+    "${S2_SOURCE_DIR}/third_party/absl/strings/match.cc"
```

Comment:
As I understand, abseil already exists in clickhouse, try to reuse this library to avoid duplicate symbols